### ISSUE TITLE
How to train the MO3TRTemp andBoxTransformer Module?

### ISSUE THREAD

**Issue by Soulmate7** [alanzty/MO3TR#4]
hello, I am interesting in your paper especially the MO3TRTemp andBoxTransformer Module. I read your code but I have some questions as follows:
1. How to train the MO3TRTemp andBoxTransformer Module? End to End or separate with the whole model?
2. How to set the loss function for these two module?
Hope to receive your reply. Thanks!

**Comment by alanzty** [comment]
Hi, thanks for you interest in this paper. As you may know, deformable transformer has an embedding input as well as a location input.

That is why we essentially have two temporal transformers for both embeddings and boxes. Embedding transformer can only be trained end-to-end. Boxtransformer can be trained both end-to-end or from ground truth boxes for trajectory forecasting. However, the idea of the paper is to train tracker end-to-end, therefore we train both transformers end-to-end to achieve final results.

**Comment by Soulmate7** [comment]
Thanks for your prompt reply! I do wonder which way to choose to train the BoxTransformer, now I see.
And could you tell me how long have you train the model? Because I rebuild these modules by my way and try to train them.
